counsel.el

```diff
@@ -4782,10 +4784,7 @@ An extra action allows to switch to the process buffer."
              counsel-esh--index-last snd))
       (counsel-shell-history
        (setq comint-input-ring-index snd
-             counsel-shell-history--index-last snd))
-      (counsel-slime-repl-history
```

Comment:
Why are you removing this case?

Reply:
Without additional hacks within `counsel.el` to advise the relevant parts of Slime (https://github.com/slime/slime/blob/15cf0609d30255405957bf0612fd6291fea438bc/contrib/slime-repl.el#L984-L985), we don't presently gain any actual benefit from including that case. With `comint` and `eshell`, however, we already have the necessary machinery in place.

Reply:
The main goal of these index reassignments was to support history replay via `C-c C-x` in `comint` and `eshell`. Slime does not presently have any history-replay command at present.

Reply:
How about, instead of removing the case, we just add a comment next to it explaining this situation, i.e. that it's currently a no-op? In the name of being conservative, rather than anything else.

Reply:
OK, I've updated the patch to do that.